daemon/containerd/service.go

```diff
@@ -139,6 +144,57 @@ func (i *ImageService) GetLayerFolders(img *image.Image, rwLayer layer.RWLayer)
 }
 
 // GetContainerLayerSize returns the real size & virtual size of the container.
-func (i *ImageService) GetContainerLayerSize(containerID string) (int64, int64) {
-	panic("not implemented")
+func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID string) (int64, int64, error) {
+	snapshotter := i.client.SnapshotService(containerd.DefaultSnapshotter)
```

Comment:
Should this use the snapshotter that was configured instead of the default (also see https://github.com/rumpl/moby/pull/76)?


(I guess _technically_  it's even possible for each container to use a different snapshotter, so could even be the snapshotter used for the container)

Reply:
I was wondering about this, but didn't say anything because I figured that we only reason in terms of one snapshotter. However, realizing that snapshotters are different from the content store, that seems naive. this should probably be `i.snapshotter` to make a future refactor with per-container snapshotters more feasible.

Reply:
Given that we load the container config, perhaps `container.Driver` would work as well; even if we currently only support a single snapshotter, doing so would make it already "future proof" for if we would support different snapshotters per container 🤔 


I just found that one of my older "upstream PR" had some changes related to that (I just rebased it, and looks like we didn't include those changes in moby yet); https://github.com/moby/moby/pull/44075

Reply:
Ah, good catch. It feels like we should probably prefer `container.Driver` in other places where we use `i.snapshotter` in the context of a single container as well.

Reply:
Looking through the code we're mostly using the configured snapshotter `i.snapshotter`, but we can use the container's I think. Would that be preferable here, or should we save that for a later pass to see everywhere where we align to use the container's configured one?

Reply:
I was leaning to the latter with the idea of changing this PR to use `i.snapshotter`, but I have no strong feelings one way or another. I suppose in a vacuum using `i.snapshotter` and leaving the introduction of per-container snapshotters for a later refactoring PR might make the lineage of the code more clear.

Reply:
Agreed, let's use `i.snapshotter`, the per-container snapshotter was already discussed last year and the idea was that we don't want to do this IIRC. If we decide to do it in the future the changes are much more bigger so let's not confuse future readers by using the container's snapshotter.

Reply:
Sounds good 👍 I updated the PR to use `i.snapshotter` :)